src/main/java/com/fasterxml/jackson/databind/deser/DeserializerCache.java

```diff
@@ -52,6 +53,12 @@ public final class DeserializerCache
     protected final HashMap<JavaType, JsonDeserializer<Object>> _incompleteDeserializers
         = new HashMap<JavaType, JsonDeserializer<Object>>(8);
 
+
+    /**
+     * We hold an explicit lock while creating deserializers to avoid creating duplicates.
+     */
+    private final ReentrantLock _incompleteDeserializersLock = new ReentrantLock();
```

Comment:
@cowtowncoder can we keep this private as it is in the PR? I want to do follow up PRs and would like to not have to keep this name. I think it too specific. I also think that protected code is just as bad as public code in forcing us to maintain compatibility across versions. Private code can be refactored without worrying about users who write their own subclasses.

Reply:
Yes, it's fine to be `private`; no benefit/need for protected/package access here.